include/aws/http/connection.h

```diff
@@ -512,6 +512,12 @@ enum aws_http_version aws_http_connection_get_version(const struct aws_http_conn
 AWS_HTTP_API
 struct aws_channel *aws_http_connection_get_channel(struct aws_http_connection *connection);
 
+/**
+ * Returns the remote endpoint of the HTTP connection.
+ */
+AWS_HTTP_API
+struct aws_byte_cursor aws_http_connection_get_remote_endpoint_address(struct aws_http_connection *connection);
```

Comment:
Yes to this! a.) The CRT's underlying transport will not always be TCP. In some cases it will be QUIC, in others it will be weird experimental stuff that aren't even using a socket api. If you return the whole struct we can convey the type of socket. We could even expand it over time to include things like MAC address as we need to without an API breakage. You could do it without an ABI break as well if you return it as ````const struct aws_socket_endpoint  *```` as well.